x-pack/plugins/cases/public/components/recent_cases/index.tsx

```diff
@@ -0,0 +1,96 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiHorizontalRule, EuiText, EuiTitle } from '@elastic/eui';
+import React, { useMemo, useState } from 'react';
+
+import * as i18n from './translations';
+import { CaseDetailsHrefSchema, LinkAnchor } from '../links';
+import { RecentCasesFilters } from './filters';
+import { RecentCasesComp } from './recent_cases';
+import { FilterMode as RecentCasesFilterMode } from './types';
+import { useCurrentUser } from '../../common/lib/kibana';
+
+export interface RecentCasesProps {
+  allCasesHref: string;
+  createCaseHref: string;
+  getCaseDetailsHref: (caseDetails: CaseDetailsHrefSchema) => string;
+  goToAllCases: (ev: React.MouseEvent) => void;
+  onCaseDetailsNavClick: (caseDetails: CaseDetailsHrefSchema) => void;
+  perPage: number;
+}
+
+const RecentCases = ({
+  allCasesHref,
+  createCaseHref,
+  getCaseDetailsHref,
+  goToAllCases,
+  onCaseDetailsNavClick,
+  perPage,
+}: RecentCasesProps) => {
+  const currentUser = useCurrentUser();
+  const [recentCasesFilterBy, setRecentCasesFilterBy] = useState<RecentCasesFilterMode>(
+    'recentlyCreated'
+  );
+
+  const recentCasesFilterOptions = useMemo(
+    () =>
+      recentCasesFilterBy === 'myRecentlyReported' && currentUser != null
+        ? {
+            reporters: [
+              {
+                email: currentUser.email,
+                full_name: currentUser.fullName,
+                username: currentUser.username,
+              },
+            ],
+          }
+        : {},
+    [currentUser, recentCasesFilterBy]
+  );
+  return (
+    <>
+      <>
+        <EuiFlexGroup alignItems="center" gutterSize="none" justifyContent="spaceBetween">
+          <EuiFlexItem grow={false}>
+            <EuiTitle size="xs">
+              <h2>{i18n.RECENT_CASES}</h2>
+            </EuiTitle>
+          </EuiFlexItem>
+
+          <EuiFlexItem grow={false}>
+            <RecentCasesFilters
+              filterBy={recentCasesFilterBy}
+              setFilterBy={setRecentCasesFilterBy}
+              showMyRecentlyReported={currentUser != null}
+            />
+          </EuiFlexItem>
+        </EuiFlexGroup>
+        <EuiHorizontalRule margin="s" />
+      </>
+      <EuiText color="subdued" size="s">
+        <RecentCasesComp
+          createCaseHref={createCaseHref}
+          filterOptions={recentCasesFilterOptions}
+          getCaseDetailsHref={getCaseDetailsHref}
+          onCaseDetailsNavClick={onCaseDetailsNavClick}
+          perPage={perPage}
+        />
+        <EuiHorizontalRule margin="s" />
+        <EuiText size="xs">
+          <LinkAnchor onClick={goToAllCases} href={allCasesHref}>
+            {' '}
+            {i18n.VIEW_ALL_CASES}
+          </LinkAnchor>
+        </EuiText>
+      </EuiText>
+    </>
+  );
+};
+
+// eslint-disable-next-line import/no-default-export
+export { RecentCases as default };
```

Comment:
Why don't we `export React.memo(RecentCases);` ?

Reply:
why tho? what benefit does wrapping it in `React.memo` give?